sockeye/encoder.py

```diff
@@ -267,16 +267,31 @@ def get_num_hidden(self):
         return self.num_hidden
 
 
-class EmbeddingConfig(Config):
+class FactorConfig(config.Config):
+
+    def __init__(self, vocab_size: int, num_embed: int) -> None:
+        super().__init__()
+        self.vocab_size = vocab_size
+        self.num_embed = num_embed
+
+
+class EmbeddingConfig(config.Config):
 
     def __init__(self,
                  vocab_size: int,
                  num_embed: int,
-                 dropout: float) -> None:
+                 dropout: float,
+                 is_source: bool = False,
```

Comment:
It feels wrong that this is part of the config, rather than being a parameter to `Embedding .__init__`. It's not something configurable and one could simply pass the parameter to the Embedding constructor in `_build_model_components`.

Reply:
Makes sense!